### ISSUE TITLE
mxRun: reports  Error in (function (cl, name, valueClass)

### ISSUE THREAD

**Issue by tbates** [OpenMx/OpenMx#166]
m2 below can be built OK, but when run emits this error:

> Error in (function (cl, name, valueClass)  : 
  assignment of an object of class “character” is not valid for @‘formula’ in an object of class “MxAlgebra”; is(value, "MxAlgebraFormula") is not TRUE

```R
#######
# setup #
#######
require(umx); data(demoOneFactor)
manifests = names(demoOneFactor)
myData    = mxData(cor(demoOneFactor), type = "cor", numObs = 500)

m1 <- umxRAM("OneFactor", data = myData,
	umxPath("g", to = manifests),
	umxPath(var = manifests),
	umxPath(var = "g", fixedAt = 1)
)

#############################################
# Problem model: breaks at run time with inscrutable error  #
#############################################
m2 <- mxModel(m1, name= "cor_ekt",
	mxMatrix(name= "I", type="Iden", ncol=length(manifestVars)),
	mxMatrix(name= "unit_1_by_nVar", type="Unit", ncol=1, nrow=length(manifestVars)),
	mxConstraint(name="constraintForCorData", diag2vec(F %*% solve(I-A) %*% S %*% t(solve(I-A)) %*% t(F)) == "unit_1_by_nVar")
)
m2 = mxRun(m2)

Error in (function (cl, name, valueClass)  : 
  assignment of an object of class “character” is not valid for @‘formula’ in an object of class “MxAlgebra”; is(value, "MxAlgebraFormula") is not TRUE

```

**Comment by mcneale** [comment]
At first I thought this might be something umx is inscrutably doing wrong. But it seems to be due to having “” around the right hand side of the constraint formula. The error could be improved; it misdirects mxAlgebra for mxConstraint ( that seems to me could be a bit of a pain to catch as constraints are understood as algebras) and it doesn’t say what the offending character string is - that may prove easier to report back to the user.  As would reporting _which_ algebra/constraint triggered the error (some more nuisance housekeeping , but seems worthwhile to avoid user panic).  This looks like an R error (sadly often gobbledegook) so it seems we need to try() again :).

**Comment by tbates** [comment]
yeah... so perhaps mxConstraint should fail, reporting the string as not a legal right hand side in a constraint algebra?

**Comment by mcneale** [comment]
Ideally, yes, but it is only when it is parsed as an algebra that the issue is discovered.  Hence the misdirect to mxAlgebra issue.

**Comment by tbates** [comment]
Seems like the `MxAlgebra.R` fn `algebraErrorChecking` (called in `mxConstraint`) should be catching that the whole rhs is a string?

So this would be a requires to enhance the checking by `algebraErrorChecking` to include not accepting strings as left or right hand sides.

currently can feed it like this and it doesn't care:

```R
tmp = mxConstraint(I-A) == "bad robot")
OpenMx:::algebraErrorChecking(tmp$formula[[3]], "mxConstraint")
```

**Comment by mcneale** [comment]
It cares about the missing ( though :).  Perhaps this 'feature' is to enable use of [ and ] element references in constraints?  I worry a lot about breaking stuff that used to work.

```R
tmp = mxConstraint((I-A) == "bad robot")
OpenMx:::algebraErrorChecking(tmp$formula[[3]], "mxConstraint")
```
